typings/gatherer.d.ts

```diff
@@ -13,10 +17,83 @@ declare global {
     export interface LoadData {
       networkRecords: Array<void>;
       devtoolsLog: Array<void>;
-      trace: {traceEvents: Array<TraceEvent>}
+      trace: {traceEvents: Array<TraceEvent>};
+    }
+
+    namespace Artifact {
+      export interface LanternMetric {
+        timing: number;
+        optimisticEstimate: Simulation.Result;
+        pessimisticEstimate: Simulation.Result;
+        optimisticGraph: Simulation.GraphNode;
+        pessimisticGraph: Simulation.GraphNode;
+      }
+
+      export interface TraceTimes {
+        navigationStart: number;
+        firstPaint: number;
+        firstContentfulPaint: number;
+        firstMeaningfulPaint: number;
+        traceEnd: number;
+        onLoad: number;
+        domContentLoaded: number;
+      }
+
+      export interface TraceOfTab {
+        timings: TraceTimes;
+        timestamps: TraceTimes;
+        processEvents: Array<TraceEvent>;
+        mainThreadEvents: Array<TraceEvent>;
+        startedInPageEvt: TraceEvent;
+        navigationStartEvt: TraceEvent;
+        firstPaintEvt: TraceEvent;
+        firstContentfulPaintEvt: TraceEvent;
+        firstMeaningfulPaintEvt: TraceEvent;
+        onLoadEvt: TraceEvent;
+        fmpFellBack: boolean;
+      }
+    }
+
+    namespace Simulation {
+      // HACK: TS treats 'import * as Foo' as namespace instead of a type, use typeof and prototype
+      export type GraphNode = typeof _Node['prototype'];
```

Comment:
2.8 ended up adding a new helper that might(?) work here (and be somewhat less ugly :). `export type GraphNode = InstanceType<typeof _Node>;`